torch/nested/_internal/union_find.py

```diff
@@ -124,6 +124,23 @@ def find(a):
     uf = get_union_find()
     return uf.find(a)
 
+lib = torch.library.Library("nested", "FRAGMENT")
+lib.define("get_max_seqlen(Tensor x) -> SymInt")
+
+def foo_impl(x):
+    uf = get_union_find()
+    cached_metadata = uf.get_metadata(x)
+    if not cached_metadata.get("_max_seqlen"):
+        cached_metadata["_max_seqlen"] = torch.max(x).item()
```

Comment:
I don't think this calculation is correct if `x` is offsets - it'd need to be a max on `x.diff()`

Reply:
Yup, good catch.